### PR TITLE
services: resolve community slugs for record search

### LINKED ISSUE #1188: Search for records of communities fails when identifier used instead of UUID
Package version (if known): InvenioRDM v9.1

## Describe the bug
Using the API to search for the records of a community by using the community identifier should return records of the community but it doesn't. 

For example, for community with identifier `123123` and UUID `a2dfb1ce-8a28-4a55-ab3f-e0873f290a30`

:x:  The following doesn't work:

`curl 'https://inveniordm-qa.web.cern.ch/api/communities/123123/records?q=&sort=newest&page=1&size=10' -H 'Accept: application/vnd.inveniordm.v1+json'` 

result:

```json
{"aggregations": {"access_status": {"buckets": [], "label": "Access status"}, "resource_type": {"buckets": [], "label": "Resource types"}}, "hits": {"hits": [], "total": 0}}
```

:heavy_check_mark: The following works:

`curl 'https://inveniordm-qa.web.cern.ch/api/communities/a2dfb1ce-8a28-4a55-ab3f-e0873f290a30/records?q=&sort=newest&page=1&size=10' -H 'Accept: application/vnd.inveniordm.v1+json'` 

result

```json
{"aggregations": {"access_status": {"buckets": [{"key": "metadata-only", "doc_count": 2, "label": "Metadata-only", "is_selected": false}], "label": "Access status"}, "resource_type": {"buckets": [{"key": "image", "doc_count": 1, "label": "Image", "is_selected": false, "inner": {"buckets": []}}, {"key": "publication", "doc_count": 1, "label": "Publication", "is_selected": false, "inner": {"buckets": []}}], "label": "Resource types"}}, "hits": {"hits": [{"access": {"record": "public", "embargo": {"reason": null, "active": false}, "status": "metadata-only", "files": "public"}, "updated": "2022-07-11T15:03:16.894556+00:00", "revision_id": 3, "created": "2022-07-11T15:03:16.796444+00:00", "is_draft": false, "parent": {"communities": {"default": "a2dfb1ce-8a28-4a55-ab3f-e0873f290a30", "ids": ["a2dfb1ce-8a28-4a55-ab3f-e0873f290a30"]}, "id": "cbawh-08e39"}, "id": "fe1pm-ka418", "pids": {"oai": {"identifier": "oai:inveniordm.web.cern.ch:fe1pm-ka418", "provider": "oai"}, "doi": {"identifier": "10.81088/fe1pm-ka418", "client": "datacite", "provider": "datacite"}}, "is_published": true, "versions": {"is_latest": true, "index": 1}, "metadata": {"title": "test", "publication_date": "2022-07-11", "publisher": "CERN", "resource_type": {"id": "publication", "title": {"en": "Publication"}}, "rights": [{"icon": "cc-by-icon", "id": "cc-by-4.0", "props": {"url": "https://creativecommons.org/licenses/by/4.0/legalcode", "scheme": "spdx"}, "description": {"en": "The Creative Commons Attribution license allows re-distribution and re-use of a licensed work on the condition that the creator is appropriately credited."}, "title": {"en": "Creative Commons Attribution 4.0 International"}}], "creators": [{"person_or_org": {"name": "dasfdasf", "type": "personal", "family_name": "dasfdasf"}}]}, "status": "published", "links": {"self": "https://inveniordm-qa.web.cern.ch/api/records/fe1pm-ka418", "self_html": "https://inveniordm-qa.web.cern.ch/records/fe1pm-ka418", "self_doi": "https://inveniordm-qa.web.cern.ch/doi/10.81088/fe1pm-ka418", "doi": "https://doi.org/10.81088/fe1pm-ka418", "self_iiif_manifest": "https://inveniordm-qa.web.cern.ch/api/iiif/record:fe1pm-ka418/manifest", "self_iiif_sequence": "https://inveniordm-qa.web.cern.ch/api/iiif/record:fe1pm-ka418/sequence/default", "files": "https://inveniordm-qa.web.cern.ch/api/records/fe1pm-ka418/files", "latest": "https://inveniordm-qa.web.cern.ch/api/records/fe1pm-ka418/versions/latest", "latest_html": "https://inveniordm-qa.web.cern.ch/records/fe1pm-ka418/latest", "draft": "https://inveniordm-qa.web.cern.ch/api/records/fe1pm-ka418/draft", "versions": "https://inveniordm-qa.web.cern.ch/api/records/fe1pm-ka418/versions", "access_links": "https://inveniordm-qa.web.cern.ch/api/records/fe1pm-ka418/access/links", "reserve_doi": "https://inveniordm-qa.web.cern.ch/api/records/fe1pm-ka418/draft/pids/doi"}, "files": {"enabled": false, "order": []}, "ui": {"is_draft": false, "publication_date_l10n_long": "July 11, 2022", "resource_type": {"id": "publication", "title_l10n": "Publication"}, "creators": {"creators": [{"person_or_org": {"name": "dasfdasf", "type": "personal", "family_name": "dasfdasf"}}], "affiliations": []}, "publication_date_l10n_medium": "Jul 11, 2022", "rights": [{"icon": "cc-by-icon", "id": "cc-by-4.0", "props": {"url": "https://creativecommons.org/licenses/by/4.0/legalcode", "scheme": "spdx"}, "description_l10n": "The Creative Commons Attribution license allows re-distribution and re-use of a licensed work on the condition that the creator is appropriately credited.", "title_l10n": "Creative Commons Attribution 4.0 International"}], "version": "v1", "updated_date_l10n_long": "July 11, 2022", "access_status": {"id": "metadata-only", "title_l10n": "Metadata-only", "description_l10n": "No files are available for this record.", "icon": "tag", "embargo_date_l10n": null, "message_class": ""}, "created_date_l10n_long": "July 11, 2022"}}, {"access": {"record": "public", "embargo": {"reason": null, "active": false}, "status": "metadata-only", "files": "public"}, "updated": "2022-07-11T14:54:30.561709+00:00", "revision_id": 3, "created": "2022-07-11T14:54:30.482669+00:00", "is_draft": false, "parent": {"communities": {"default": "a2dfb1ce-8a28-4a55-ab3f-e0873f290a30", "ids": ["a2dfb1ce-8a28-4a55-ab3f-e0873f290a30"]}, "id": "5kt1d-y3670"}, "id": "7kfy5-rbn51", "pids": {"oai": {"identifier": "oai:inveniordm.web.cern.ch:7kfy5-rbn51", "provider": "oai"}, "doi": {"identifier": "10.81088/7kfy5-rbn51", "client": "datacite", "provider": "datacite"}}, "is_published": true, "versions": {"is_latest": true, "index": 3}, "metadata": {"description": "<p>test</p>", "title": "Record belongs to Javier's community edited new version!", "publication_date": "1996", "publisher": "CERN", "resource_type": {"id": "image", "title": {"en": "Image"}}, "rights": [{"icon": "cc-by-icon", "id": "cc-by-4.0", "props": {"url": "https://creativecommons.org/licenses/by/4.0/legalcode", "scheme": "spdx"}, "description": {"en": "The Creative Commons Attribution license allows re-distribution and re-use of a licensed work on the condition that the creator is appropriately credited."}, "title": {"en": "Creative Commons Attribution 4.0 International"}}], "creators": [{"person_or_org": {"name": "sadsad", "type": "personal", "family_name": "sadsad"}}]}, "status": "published", "links": {"self": "https://inveniordm-qa.web.cern.ch/api/records/7kfy5-rbn51", "self_html": "https://inveniordm-qa.web.cern.ch/records/7kfy5-rbn51", "self_doi": "https://inveniordm-qa.web.cern.ch/doi/10.81088/7kfy5-rbn51", "doi": "https://doi.org/10.81088/7kfy5-rbn51", "self_iiif_manifest": "https://inveniordm-qa.web.cern.ch/api/iiif/record:7kfy5-rbn51/manifest", "self_iiif_sequence": "https://inveniordm-qa.web.cern.ch/api/iiif/record:7kfy5-rbn51/sequence/default", "files": "https://inveniordm-qa.web.cern.ch/api/records/7kfy5-rbn51/files", "latest": "https://inveniordm-qa.web.cern.ch/api/records/7kfy5-rbn51/versions/latest", "latest_html": "https://inveniordm-qa.web.cern.ch/records/7kfy5-rbn51/latest", "draft": "https://inveniordm-qa.web.cern.ch/api/records/7kfy5-rbn51/draft", "versions": "https://inveniordm-qa.web.cern.ch/api/records/7kfy5-rbn51/versions", "access_links": "https://inveniordm-qa.web.cern.ch/api/records/7kfy5-rbn51/access/links", "reserve_doi": "https://inveniordm-qa.web.cern.ch/api/records/7kfy5-rbn51/draft/pids/doi"}, "files": {"enabled": false, "order": []}, "ui": {"description_stripped": "test", "is_draft": false, "publication_date_l10n_long": "1996", "resource_type": {"id": "image", "title_l10n": "Image"}, "creators": {"creators": [{"person_or_org": {"name": "sadsad", "type": "personal", "family_name": "sadsad"}}], "affiliations": []}, "publication_date_l10n_medium": "1996", "rights": [{"icon": "cc-by-icon", "id": "cc-by-4.0", "props": {"url": "https://creativecommons.org/licenses/by/4.0/legalcode", "scheme": "spdx"}, "description_l10n": "The Creative Commons Attribution license allows re-distribution and re-use of a licensed work on the condition that the creator is appropriately credited.", "title_l10n": "Creative Commons Attribution 4.0 International"}], "version": "v3", "updated_date_l10n_long": "July 11, 2022", "access_status": {"id": "metadata-only", "title_l10n": "Metadata-only", "description_l10n": "No files are available for this record.", "icon": "tag", "embargo_date_l10n": null, "message_class": ""}, "created_date_l10n_long": "July 11, 2022"}}], "total": 2}}
```

## Expected behavior
It should work with both.

### PR DESCRIPTION
**Pull request by slint**
* Closes #1188.

:heart: Thank you for your contribution!

### Description

Please describe briefly your pull request.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: invenio_rdm_records/services/services.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ def search_community_records(
     ):
         """Search for records published in the given community."""
         self.require_permission(identity, "read")
-        current_communities.service.record_cls.pid.resolve(
+        community = current_communities.service.record_cls.pid.resolve(
```
Comment (ppanero) [inline invenio_rdm_records/services/services.py:132]: note for reviewers: the resolver is able to treat both slugs and uuids, see https://github.com/inveniosoftware/invenio-communities/blob/master/invenio_communities/communities/records/systemfields/pidslug.py#L31

**Review by ppanero — APPROVED** [review]
We need to consider if this needs to be backported to 9.1

**Review by ntarocco — APPROVED** [review]

**Comment by ntarocco** [comment]
> We need to consider if this needs to be backported to 9.1

@fenekku the issue was opened by you, how important is this for your use cases in LTS 9.1?
@ppanero be aware that we cannot cherry-pick the commit, we have refactored resources/services. We will need to copy/paste the fix.

**Comment by fenekku** [comment]
This fix was desired by one of our librarians who is performing automated checks of the communities we imported. We only know the community identifiers in advance.    

Thanks for the fix. At this point, we've moved on to v11 because features we need are there. So, for us, as long as it goes in a patch for v11 we are good.

**Pull request merged**